test/end2end.py

```diff
@@ -204,3 +213,56 @@ def test_hanging_fatal(self):
 
         regex = re.compile("mympirun has been running for .* seconds without seeing any output.")
         self.assertTrue(len(regex.findall(out)) == 1, "Pattern '%s' found in: %s" % (regex.pattern, out))
+
+
+    def test_option_double(self):
+        """Test --double command line option"""
+        install_fake_mpirun('mpirun', self.tmpdir, txt=FAKE_MPIRUN_MACHINEFILE)
+        cmd = "%s %s --setmpi impirun --double hostname"
+        ec, out = run_simple(cmd % (sys.executable, self.mympiscript))
+        # set_pbs_env() sets 2 cores, so double is 4
+        self.assertEqual(len(out.split('\n')), 4)
+        self.assertEqual(out, ('\n'.join(['localhost'] * 4)))
+
+
+    def test_option_hybrid(self):
+        """Test --hybrid command line option"""
+        install_fake_mpirun('mpirun', self.tmpdir, txt=FAKE_MPIRUN_MACHINEFILE)
+        ec, out = run_simple("%s %s --setmpi impirun --hybrid 5 hostname" % (sys.executable, self.mympiscript))
+        self.assertEqual(len(out.split('\n')), 5)
+        self.assertEqual(out, ('\n'.join(['localhost'] * 5)))
+
+
+    def test_option_universe(self):
+        """Test --universe command line option"""
+        install_fake_mpirun('mpirun', self.tmpdir)
+
+        self.change_env(5)
+        ec, out = run_simple("%s %s --setmpi impirun --universe 4 hostname" % (sys.executable, self.mympiscript))
+        regex = re.compile('-np 4')
+        self.assertTrue(regex.search(out))
+
+        self.change_env(1)
+        # intel mpi without hydra
+        ec, out = run_simple("%s %s --setmpi impirun --universe 1 hostname" % (sys.executable, self.mympiscript))
+        np_regex = re.compile('-np 1')
+        ncpus_regex = re.compile('--ncpus=1')
+        self.assertTrue(np_regex.search(out))
+        self.assertTrue(ncpus_regex.search(out))
+
+        # intel mpi with hydra
+        del os.environ['I_MPI_PROCESS_MANAGER']
+        ec, out = run_simple("%s %s --setmpi ihmpirun --universe 1 hostname" % (sys.executable, self.mympiscript))
+        self.assertTrue(np_regex.search(out))
+        self.assertFalse(ncpus_regex.search(out))
+
+        # re-set pbs environment
+        set_PBS_env()
+
+    def change_env(self, cores):
+        """Helper method for changing the number of cores in the machinefile"""
+        pbsnodefile = tempfile.NamedTemporaryFile(delete=False)
+        pbsnodefile.write('\n'.join(['localhost'] * cores))
+        pbsnodefile.close()
+        os.environ['PBS_NODEFILE'] = pbsnodefile.name
+        os.environ['I_MPI_PROCESS_MANAGER'] = 'mpd'
```

Comment:
@Caylo move this out of this method, so you can reuse this method later without checking in legacy (non-hydra) mode